src/linear_algebra/matrix/determinant.lean

```diff
@@ -628,6 +628,33 @@ begin
       rw [hx, from_blocks_apply₂₁], refl }}
 end
 
+/-- The determinant of a 2x2 block matrix with the upper-right block equal to zero is the product of
+the determinants of the diagonal blocks. For the generalization to any number of blocks, see
+`matrix.det_of_lower_triangular`. -/
+lemma lower_two_block_triangular_det
+  (A : matrix m m R) (C : matrix n m R) (D : matrix n n R) :
+  (matrix.from_blocks A 0 C D).det = A.det * D.det :=
+by rw [←det_transpose, from_blocks_transpose, transpose_zero, upper_two_block_triangular_det,
+  det_transpose, det_transpose]
+
+/-- A special case of the **Matrix determinant lemma** for when `A = I`.
+
+TODO: show this more generally. -/
+lemma det_one_add_col_mul_row {u v : m → R} :
```

Comment:
I would prefer to have `u` and `v` explicit. It makes it easier to perform forward-proving steps like
```
have := det_one_add_col_mul_row u v
```
in other proofs.

Reply:
This was accidental anyway.